ts/private/ts_project.bzl

```diff
@@ -248,25 +224,32 @@ This is an error because Bazel does not run actions unless their outputs are nee
                 verb = "Transpiling & type-checking"
             else:
                 verb = "Transpiling"
-        else:
-            verb = "Type-checking"
-
-        ctx.actions.run(
-            executable = executable,
-            inputs = inputs_depset,
-            arguments = [arguments],
-            outputs = outputs,
-            mnemonic = "TsProject",
-            execution_requirements = execution_requirements,
-            progress_message = "%s TypeScript project %s [tsc -p %s]" % (
-                verb,
-                ctx.label,
-                tsconfig_path,
-            ),
-            env = {
-                "BAZEL_BINDIR": ctx.bin_dir.path,
-            },
-        )
+
+    # Validation actions still need to produce some output, so we output a .validation file
+    # that ends up in the _validation output group.
+    if supports_workers:
+        run_cmd = "{} $@".format(executable.path)
+        arguments.add_all(["--bazelValidationFile", validation_output.short_path])
+    else:
+        run_cmd = """{} $@ && echo "" > {} """.format(executable.path, validation_output.path)
```

Comment:
instead of `echo ""` could we just use https://github.com/aspect-build/rules_js/blob/main/js/private/js_binary.sh.tpl#L22